File: Ad-Hoc/pagina33/3315/3315.py
def converter_para_binario(n):
    binario = ""
    while n > 0:  
        binario += str(n % 2)
        n //= 2
    return binario[::-1] if binario else "0"

semanas = 4
mnv = 0 #maior_num_visuizacoes
for i in range(semanas):
    visualizacoes = [int(x) for x in input().split()]
    soma = 0
    for dia in range(7):
        soma += visualizacoes[dia]
    if soma > mnv:
        mnv = soma
        
print(f"{mnv} = {converter_para_binario(mnv)}")

File: Ad-Hoc/pagina33/3315/3315.js

const fs = require("fs")
const input = fs.readFileSync("/dev/stdin", "utf8").split("\n")

function converterParaBinario(n) {
    let binario = "";
    while (n > 0) {
        binario += (n % 2).toString();
        n = Math.floor(n / 2);
    }
    return binario ? binario.split("").reverse().join("") : "0";
}
let mnv = 0;
for(let semana = 0; semana < 4; semana++){
    const visualizacoes = input[i].split(" ").map(Number)
    let soma = 0
    for(let dia = 0; dia < 7; dia++){
        soma += visualizacoes[dia]
    }
    if (soma > mnv) { mnv = soma }
}

console.log(`${mnv} = ${converterParaBinario(mnv)}`)